hopdrops/hopdrops
# feat: Add lambda OAuth function and API Gateway route

**Description**

This pull request contains an AWS Lambda function and AWS API Gateway route for granting our application [permission](https://developer.squareup.com/docs/api/oauth#navsection-authorizationbasics) to issue requests to the Square API.

**Verification Instructions**

0. `git checkout lambda`
1. `cd hopdrops/square/functions`
2. `zip functions.zip *.py`
3. `terraform apply -var 'zip_file=functions.zip' -var 'client_id=<square-client-id>'`
4. Click the `invoke_url` from the Terraform output:

<img width="682" alt="Screen Shot 2019-06-21 at 10 15 50 AM" src="https://user-images.githubusercontent.com/2184329/59928873-a37ee980-940d-11e9-8828-e4686982e5ee.png">

After clicking the `invoke_url`, the following page should be displayed:

<img width="1053" alt="Screen Shot 2019-06-21 at 10 17 28 AM" src="https://user-images.githubusercontent.com/2184329/59928971-d9bc6900-940d-11e9-998b-e274064e11ca.png">

**Note:** The page above requires the user to be authenticated to Square.
